test/claim.test.ts

```diff
@@ -207,4 +183,81 @@ describe('StakingPool.claim', () => {
     });
   });
 
+  context('rewardPerSecond is changed', async () => {
+    beforeEach('init the pool and stake in pool', async () => {
+      await testEnv.rewardAsset.connect(deployer).faucet();
+      await testEnv.rewardAsset.connect(deployer).approve(testEnv.stakingPool.address, RAY);
+      await testEnv.stakingPool
+        .connect(deployer)
+        .initNewPool(rewardPersecond, firstRoundStartTimestamp, duration);
+      await testEnv.stakingAsset.connect(alice).faucet();
+      const tx = await testEnv.stakingAsset.connect(alice).approve(testEnv.stakingPool.address, RAY);
+      await advanceTimeTo(await getTimestamp(tx), firstRoundStartTimestamp);
+      await testEnv.stakingPool.connect(alice).stake(amount);
+      await testEnv.rewardAsset.connect(deployer).faucet();
+      await testEnv.rewardAsset.connect(deployer).transfer(testEnv.stakingPool.address, ethers.utils.parseEther('100'));
+    });
+
+    it('rewardPerSecond is changed and claim', async () => {
+      const poolDataBefore = await getPoolData(testEnv);
+      const userDataBefore = await getUserData(testEnv, alice);
+      const tx = await testEnv.stakingPool.connect(deployer).extendPool(rewardPersecond, duration);
+
+      const [expectedPoolData_1, expectedUserData_1] = updatePoolData(
+        poolDataBefore,
+        userDataBefore,
+        await getTimestamp(tx),
+        duration,
+        rewardPersecond
+      );
+
+      const claimTx = await testEnv.stakingPool.connect(alice).claim();
+      const [expectedPoolData_2, expectedUserData_2] = expectDataAfterClaim(
+        expectedPoolData_1,
+        expectedUserData_1,
+        await getTimestamp(claimTx)
+      );
+
+      const poolDataAfter = await getPoolData(testEnv);
+      const userDataAfter = await getUserData(testEnv, alice);
+
+      expect(poolDataAfter).to.be.equalPoolData(expectedPoolData_2);
+      expect(userDataAfter).to.be.equalUserData(expectedUserData_2);
+    });
+
+    it('rewardPerSecond is changed and stake and claim', async () => {
```

Comment:
I'm not quite clear what the purpose of this test case. I think it's ok to be separated.